homeassistant/components/media_player/webostv.py

```diff
@@ -56,32 +56,37 @@
     vol.Optional(CONF_HOST): cv.string,
     vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
     vol.Optional(CONF_ON_ACTION): cv.SCRIPT_SCHEMA,
-    vol.Optional(CONF_TIMEOUT, default=8): cv.positive_int,
+    vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): cv.positive_int,
 })
 
 
 # pylint: disable=unused-argument
 def setup_platform(hass, config, add_devices, discovery_info=None):
     """Set up the LG WebOS TV platform."""
     if discovery_info is not None:
-        host = urlparse(discovery_info[1]).hostname
+        host = discovery_info.get('host')
+        name = discovery_info.get('name', DEFAULT_NAME)
+        customize = {}
+        timeout = DEFAULT_TIMEOUT
+        turn_on_action = None
+        config_file_path = WEBOSTV_CONFIG_FILE
     else:
         host = config.get(CONF_HOST)
+        name = config.get(CONF_NAME)
+        customize = config.get(CONF_CUSTOMIZE)
+        timeout = config.get(CONF_TIMEOUT)
+        turn_on_action = config.get(CONF_ON_ACTION)
+        config_file_path = config.get(CONF_FILENAME)
```

Comment:
These 5 useless lines from above are another reason why discovery should also pass the original config. I could've take them directly from there instead setting the default values already mentioned in schema :(